magic-modules-plugin/src/main/kotlin/io/labs/dotanuki/magicmodules/internal/ProjectStructureParser.kt

```diff
@@ -1,14 +1,16 @@
 package io.labs.dotanuki.magicmodules.internal
 
+import io.labs.dotanuki.magicmodules.MagicModulesExtension
 import io.labs.dotanuki.magicmodules.internal.model.GradleBuildScript
+import io.labs.dotanuki.magicmodules.internal.model.GradleFoundModule
 import io.labs.dotanuki.magicmodules.internal.model.GradleModuleType
 import io.labs.dotanuki.magicmodules.internal.model.GradleProjectStructure
 import io.labs.dotanuki.magicmodules.internal.util.e
 import io.labs.dotanuki.magicmodules.internal.util.i
 import io.labs.dotanuki.magicmodules.internal.util.logger
 import java.io.File
 
-internal class ProjectStructureParser {
+internal class ProjectStructureParser(private val magicModulesExtension: MagicModulesExtension) {
```

Comment:
I will prefer not depend on the extension here. Could you please change this constructor and pass the list of raw plugins directly?

I know we'll have to wire things at plugin entry point level, but I think this abstraction becomes better!

Reply:
Alright. I'll change.

Reply:
I created a data class to parse the raw contents to Parser. Reasons:
1. Avoid `ProjectStructureParser` constructor with 3 parameters of the same type;
2. Avoid constructor to have long parameter list;
3. Improve maintenance.